ansible/roles/test/files/ptftests/populate_fdb.py

```diff
@@ -0,0 +1,173 @@
+
+            Args:
+                None
+
+            Returns:
+                None
+        '''
+        self.dataplane = ptf.dataplane_instance
+        self.dataplane.flush()
+
+        self.testParams = testutils.test_params_get()
+        self.packetCount = self.testParams['packet_count']
+        self.startMac = self.testParams['start_mac']
+
+        self.configFile = self.testParams['config_data']
+        with open(self.configFile) as fp:
+            self.configData = json.load(fp)
+
+        self.dutMac = self.configData['dut_mac']
+        self.macToIpRatio = [int(i) for i in self.testParams['mac_to_ip_ratio'].split(':')]
+        self.assertTrue(
+            len(self.macToIpRatio) == 2 and self.macToIpRatio[0] > 0 and self.macToIpRatio[1] > 0,
+            'Invalid MAC to IP ratio: {0}'.format(self.testParams['mac_to_ip_ratio'])
+        )
+
+        if config["log_dir"] is not None:
+            filename = os.path.join(config["log_dir"], str(self)) + ".pcap"
+            self.dataplane.start_pcap(filename)
+
+    def tearDown(self):
+        '''
+            Tears down FDB instance data
+
+            Args:
+                None
+
+            Returns:
+                None
+        '''
+        if config["log_dir"] is not None:
+            self.dataplane.stop_pcap()
+
+    def __convertMacToInt(self, mac):
+        '''
+            Converts MAC address to integer
+
+            Args:
+                mac (str): MAC Address
+
+            Returns:
+                mac (int): integer representation of MAC address
+        '''
+        return int(mac.translate(None, ":.- "), 16)
+
+    def __convertMacToStr(self, mac):
+        '''
+            Converts MAC address to string
+
+            Args:
+                mac (int): MAC Address
+
+            Returns:
+                mac (str): string representation of MAC address
+        '''
+        mac = "{:012x}".format(mac)
+        return ":".join(mac[i : i + 2] for i in range(0, len(mac), 2))
+
+    def __prepareVmIp(self):
+        '''
+            Prepares VM IP addresses
+
+            Args:
+                None
+
+            Returns:
+                vmIp (dict): Map containing vlan to VM IP address
+        '''
+        vmIp = {}
+        for vlan, config in self.configData['vlan_interfaces'].items():
+            prefixLen = self.configData['vlan_interfaces'][vlan]['prefixlen']
+            ipCount = 2**(32 - prefixLen) - 3
+            numDistinctIp = self.packetCount * self.macToIpRatio[1] / self.macToIpRatio[0]
+            self.assertTrue(
+                ipCount >= numDistinctIp,
+                'Vlan network \'{0}\' does not support the requested number of IPs \'{1}\''.format(
+                    ipCount,
+                    numDistinctIp
+                )
+            )
+            vmIp[vlan] = ipaddress.ip_address(unicode(config['addr'])) + 1
+
+        return vmIp
+
+    def __populateDutFdb(self):
+        '''
+            Populates DUT FDB entries
```

Comment:
Can you describe this method a bit more? some insight into what is actually being done would be helpful